### ISSUE TITLE
Iosevka font glyphs broken

### ISSUE THREAD

**Issue by jmederosalvarado** [ryanoasis/nerd-fonts#754]
#### 🎯 Subject of the issue

I just downloaded Iosevka from the patched-fonts folder and I'm getting this weird glyph size. It's like they scaled incorrectly. It looks to me as if they were monospaced but without being centered. I'm trying to use the non-mono version, so I'm expecting glyphs to have double width.

#### 🔧 Your Setup

- `Iosevka Nerd Font Complete Windows Compatible.ttf`)_
- _Windows Terminal_
- _Windows11_

#### ★ Screenshots (Optional)

![Screenshot (44)](https://user-images.githubusercontent.com/46798594/148454219-738a1c6c-78f6-4504-98fd-06e447199775.png)
![Screenshot (45)](https://user-images.githubusercontent.com/46798594/148454223-a6f4c146-84c9-4685-92bf-d35bc5b99798.png)

**Comment by Finii** [comment]
Glyphs
* `U_E34B`  (different from report, but close enough ;)
* `U_E714` 
* `U_F0E7` 
* `U_E0B6` 

Did not find this earth with moon ;) glyph

![image](https://user-images.githubusercontent.com/16012374/148518889-bab20c54-d62e-4074-b4e9-3bf6d1be4e3d.png)

All glyphs in font are correct (wider than one advance width (500), because this is the non-mono font):

![image](https://user-images.githubusercontent.com/16012374/148519834-fce8bdb4-3647-4979-ae43-8927ce312ad6.png)

* `U_E34B`  1240
* `U_E714`  698
* `U_F0E7`  500
* `U_E0B6`  590

Although they are in an 'Ambiguous' range (https://github.com/microsoft/terminal/blob/main/src/types/CodepointWidthDetector.cpp#L246) Terminal rescales the glyphs as one-width.

I do not want to additionally also dive into _Terminal_'s code :-( and why it thinks some glyphs have to be scaled down.

**Comment by Finii** [comment]
Will be fixed by a combination of
* #748
* https://github.com/ryanoasis/nerd-fonts/issues/731#issuecomment-1003071050

Maybe we should have a dedicated Issue for 'need non-mono with mono advance' (i.e. `Nerd Font MonoBigIcons`) as a lot people want that and not the `Nerd Font Mono` but `Nerd Font (nonmono)` does not work for them.

And `Nerd Font (nonmono)` used to be `Nerd Font MonoBigIcons` ... until this strangely nonreferenced and bad documented Arch Issue *sigh*

**Comment by Finii** [comment]
Forgot the image:

![image](https://user-images.githubusercontent.com/16012374/148525619-089873e8-c599-4661-8fab-f7918df5ce99.png)

Created in this way:
```
$ git checkout feature/allow-downscaling-nonmono
$ echo move one line, see following diff
$ fontforge font-patcher --complete --windows src/unpatched-fonts/Iosevka/Regular/iosevka-regular.ttf
```
```diff
diff --git a/font-patcher b/font-patcher
index f9c17689..c73337cd 100755
--- a/font-patcher
+++ b/font-patcher
@@ -817,11 +817,11 @@ class font_patcher:
             # also ensures the font is considered monospaced on Windows by setting the
             # same width for all character glyphs. This needs to be done for all glyphs,
             # even the ones that are empty and didn't go through the scaling operations.
-            self.set_glyph_width_mono(self.sourceFont[currentSourceFontGlyph])
 
             # Ensure after horizontal adjustments and centering that the glyph
             # does not overlap the bearings (edges)
             self.remove_glyph_neg_bearings(self.sourceFont[currentSourceFontGlyph])
+            self.set_glyph_width_mono(self.sourceFont[currentSourceFontGlyph])
 
         # end for
 
```

**Comment by jmederosalvarado** [comment]
> All glyphs in font are correct (wider than one advance width (500), because this is the non-mono font):
> 
> ....
> 
> * `U_E34B`  1240
> * `U_E714`  698
> * `U_F0E7`  500
> * `U_E0B6`  590
>
> Although they are in an 'Ambiguous' range (https://github.com/microsoft/terminal/blob/main/src/types/CodepointWidthDetector.cpp#L246) Terminal rescales the glyphs as one-width.

So the problem is in the windows terminal, not nerd fonts. Right?

**Comment by jmederosalvarado** [comment]
> Forgot the image:
> 
> ![image](https://user-images.githubusercontent.com/16012374/148525619-089873e8-c599-4661-8fab-f7918df5ce99.png)
> 
> Created in this way:
> ```
> $ git checkout feature/allow-downscaling-nonmono
> $ echo move one line, see following diff
> $ fontforge font-patcher --complete --windows src/unpatched-fonts/Iosevka/Regular/iosevka-regular.ttf
> ```
> ```diff
> diff --git a/font-patcher b/font-patcher
> index f9c17689..c73337cd 100755
> --- a/font-patcher
> +++ b/font-patcher
> @@ -817,11 +817,11 @@ class font_patcher:
>              # also ensures the font is considered monospaced on Windows by setting the
>              # same width for all character glyphs. This needs to be done for all glyphs,
>              # even the ones that are empty and didn't go through the scaling operations.
> -            self.set_glyph_width_mono(self.sourceFont[currentSourceFontGlyph])
>  
>              # Ensure after horizontal adjustments and centering that the glyph
>              # does not overlap the bearings (edges)
>              self.remove_glyph_neg_bearings(self.sourceFont[currentSourceFontGlyph])
> +            self.set_glyph_width_mono(self.sourceFont[currentSourceFontGlyph])
>  
>          # end for
>  
> ```

I tested patching with this branch and still got the same result. Maybe I'm missing something

**Comment by Finii** [comment]
The branch and additionaly edit `font-patcher` like shown in the diff above?

**Comment by Finii** [comment]
i.e. move the line with `set_glyph_wifth_mono()` behind `remove_glyph_neg_bearings()`...

**Comment by jmederosalvarado** [comment]
Worked like a charm. Thank you so much for your help.

**Comment by Finii** [comment]
I reopen this, because - although it is fixed for you - it is not in general, and I wanted to create a MR ;)

**Comment by Chaoschaot232** [comment]
I hope that this already existing Bug Ticket arround this Font suites my problem:

![image](https://user-images.githubusercontent.com/1703441/161432297-f6d08125-c3f6-40ec-a8ba-0f1014f34e26.png)

OS: Win 11 Pro - Evaluation Copy - Build 2258.ni_release.220318-1623

**Comment by Finii** [comment]
> I hope that this already existing Bug Ticket arround this Font suites my problem:

I guess #694 is your issues.

I fixed the bug in `fontforge` but `nerd-fonts` does not use the new `ff` release yet. You need to self patch with a Match 2022 fontforge, or I need to upload a newly patched font set for you somewhere.

**Comment by Chaoschaot232** [comment]
Eiter this or I am just fine with it :)

Sidenote: Maybe you can edit your script slightly so that it automaticly overwrites or not already existing fonts (PS Script for windows)

**Comment by Finii** [comment]
Please raise a (new) feature request / issue about the PS script; here it will be lost.

**Comment by kaykhancheckpoint** [comment]
I have the same problem the icons seem super small now. Any fix for this?

**Comment by Finii** [comment]
@kaykhancheckpoint 
Screenshot? Which OS and console / terminal application? Which font (filename) downloaded from where (link)?
Some fixes are given above, tried any?

**Comment by saumyajyoti** [comment]
Facing the issue with latest 2.2 iosevka drop.

**Comment by Finii** [comment]
> Facing the issue with latest 2.2 iosevka drop. Windows terminal on WIndows 11.

Could you provide a screenshot?
And a screenshot of the settings? 

Which of the many Iosevkas do you use?

Which version? 2.2.1?

**Comment by saumyajyoti** [comment]
![image](https://user-images.githubusercontent.com/27477973/187199471-9b53a439-7d78-4d73-ab8f-a1558e355ec0.png)
I tried 2.2.1. The powerline symbols are better now, but still the glyphs are shown very small.
![image](https://user-images.githubusercontent.com/27477973/187199616-c79aafdc-aca9-4fe2-b3ce-57943e5885f2.png)

The Terminal with v 2.2.1:
![image](https://user-images.githubusercontent.com/27477973/187200436-d5754079-5bd1-4d6a-848c-23cd592e9139.png)

Terminal with old 2.1 font.
![image](https://user-images.githubusercontent.com/27477973/187201057-68a4ff48-bcb3-4102-bccf-8427439b7a69.png)

**Comment by Finii** [comment]
> but still the glyphs are shown very small.

See https://github.com/ryanoasis/nerd-fonts/issues/898

I do not know how Nerd Fonts should proceed here.

**Comment by saumyajyoti** [comment]
Thanks for quick update. Yes, looks like option 3 mentioned there would be needed for terminal. As there is a term variant of the font, can that variant use option 3?

**Comment by saumyajyoti** [comment]
Also, as there are varied requirements, maybe it would be better if there can be reference to patcher command line to be used for each of the use cases/terminals. Then we can easily patch the fonts as per need.

**Comment by Finii** [comment]
Ah, a good idea to make that a command line switch for `font-patcher`. There is one for `Mono`, but one can not decide between the other two. This would be a good first step :+1:

**Comment by Finii** [comment]
Implemented in `v2.2.2`.
